pyguitemp/threads.py

```diff
@@ -185,3 +187,58 @@ def abort_thread(self, ident: int) -> None:
     def post_event(self, event: ThreadResult):
         """Adds an event to the event loop of the main thread."""
         wx.PostEvent(self._window, event)
+
+
+def run_in_thread(
+    target: Callable,
+    on_abort: Optional[Callable] = None,
+    on_complete: Optional[Callable] = None,
+    on_error: Optional[Callable] = None,
+    daemon: Optional[bool] = None,
+) -> int:
+    """Is an alias for ThreadPool().run_thread(...)."""
+    return ThreadPool().run_thread(target, on_abort, on_complete, on_error, daemon)
+
+
+def run_daemon(
+    target: Callable,
+    on_abort: Optional[Callable] = None,
+    on_complete: Optional[Callable] = None,
+    on_error: Optional[Callable] = None,
+) -> int:
+    """Is an alias for ThreadPool().run_thread(..., daemon=True)."""
+    return ThreadPool().run_thread(target, on_abort, on_complete, on_error, daemon=True)
+
+
+def abort_thread(ident: int) -> None:
+    """Set thread with given identifier to abort.
+
+    Args:
+        ident: Thread identifier
+
+    Raises:
+        KeyError: If the identifier is not in the ThreadPool
+    """
+    if ident in ThreadPool()._workers:
+        ThreadPool().abort_thread(ident)
+    else:
+        key_err = f"Thread with index: {ident} is not in the ThreadPool."
+        logger.exception(key_err)
+        raise KeyError(key_err)
+
+
+def should_abort(ident: int) -> bool:
+    """Return whether the thread with given identifier should abort or not.
+
+    Args:
+        ident: Thread identifier
+
+    Raises:
+        KeyError: If the identifier is not in the ThreadPool
+    """
+    if ident in ThreadPool()._workers:
```

Comment:
As above, this check should be done within `ThreadPool().query_abort`, so both functions behave in the same whay. This is particularly important in this case because `ThreadPool().query_abort` does not take any input argument while your implementation of `should_about` does.

Reply:
That brings to me a question regarding the design choice. 

Choosing between passing `ident` explicitly or the existing implementation ` self._workers[threading.get_ident()].abort` that internally gets the identifier based on the thread that is calling.

 If the second option (existing implementation) is the preferred choice, how should we design the alias?. Also, this will be different to the `abort_thread` function that explicitly takes the `ident`.
 
There will be an explicit `ident` as an argument even in `query_abort` and will be similar to `should_abort`, in the case of the first choice. 

I am going ahead with the first choice, explicit `ident` in the changes. But depending on our final selection, I will revise it, if needed.

Reply:
The key point - which is not mentioned anywhere and should be explained - is that `abort_thread` is meant to be called by the main thread or any thread other than the one to be aborted, and so it needs as input the id of that thread. 

`query_abort`, on the contrary, is meant to be called _within the thread that, possibly, needs aborting_. In this case, the id of the thread can be atomaticlaly retrieved and does not need to be provided externally.

Threads, contrary to processes, cannot be killed (I'm talking about python here). The only way of finishing a thread is for the target function to return. `query_abort` is there so the target function can check from time to time if it needs aborting, and do so if required.

In summary: 
- `should_abort` must have `ident` as input.
- `query_abort` must have no inputs.